pkg/aws/ec2/ec2.go

```diff
@@ -691,6 +694,51 @@ func (c *Client) UnassignENIPrefixes(ctx context.Context, eniID string, prefixes
 	return err
 }
 
+func (c *Client) AssociateEIP(ctx context.Context, instanceID string, eipTags ipamTypes.Tags) (string, error) {
+	filters := make([]ec2_types.Filter, 0, len(eipTags))
+	for k, v := range eipTags {
+		filters = append(filters, ec2_types.Filter{
+			Name:   aws.String(fmt.Sprintf("tag:%s", k)),
+			Values: []string{v},
+		})
+	}
+
+	describeAddressesInput := &ec2.DescribeAddressesInput{
+		Filters: filters,
```

Comment:
What happens when a user doesn't configure filters ?

Does we end up listing all addresses ? Should we add a check to mandate filters ?

Some users might end up accidentally making them too wide as well. Does this paginate automatically ?

Reply:
My idea was that the `AssociateEIP` function will never be called if there are no filters: https://github.com/DataDog/cilium/blob/0c522d26fac45616decd34baf5c5dafa3c6377d3/pkg/ipam/node.go#L900

However, we could also add a safeguard in the function itself to prevent it from being called with empty filters (I think it's good practice to force users to have them)

That's a good point regarding pagination, I'll need to give it a try.

Reply:
I also verified, this API action is not paginated in the Go client unfortunately: https://github.com/aws/aws-sdk-go-v2/blob/service/ec2/v1.176.0/service/ec2/api_op_DescribeAddresses.go 

There is no `MaxResults` or anything like that - compare this with https://github.com/aws/aws-sdk-go-v2/blob/service/ec2/v1.176.0/service/ec2/api_op_DescribeAddressesAttribute.go for example

Reply:
It's not only the Go SDK which doesn't support it but it's not even supported server-side: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAddresses.html

I suppose it's because by default an AWS account can have [only 5 EIPs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-limit) so they probably don't think it's a resource which warrants pagination support.